scipy/optimize/_zeros_py.py

```diff
@@ -1393,3 +1394,222 @@ def toms748(f, a, b, args=(), k=1,
                           maxiter=maxiter, disp=disp)
     x, function_calls, iterations, flag = result
     return _results_select(full_output, (x, function_calls, iterations, flag))
+
+
+def chandrupatla(f, x0, x1, args=(), xtol=_xtol,
```

Comment:
In https://github.com/scipy/scipy/issues/7242#issuecomment-1370156859, I described a separate `rhs` argument that can improve performance when `f` is always the same function just with different right-hand sides. Please ignore that in the initial PR; we can include it as a follow-up.

Reply:
Sure, got it.